### PR TITLE
Swagger API functionality

### PR DESCRIPTION
**Pull request by Alankrita08**
This PR introduces Swagger API functionality to the Hardhat Website project using DRF-YASG. It enhances API documentation, enabling real-time API discovery and interactive testing via Swagger UI.

### Key updates:
Integrated DRF-YASG for Swagger documentation.
Configured URLs and settings to support Swagger.
Added metadata and grouped APIs for structured navigation.
Resolved migration conflicts to ensure seamless integration.

### Testing Done:
Verified Swagger UI functionality at /swagger/.
Tested all key endpoints for accuracy and responsiveness.
Addressed and resolved related migration conflicts.

An evidence document detailing the implementation process, testing results is included:
[Swagger API Implementation Evidence Document.pdf](https://github.com/user-attachments/files/18046584/Swagger.API.Implementation.Evidence.Document.pdf)

### CODE REVIEW

File: core/settings.py [hunk 1]
```diff
@@ -64,8 +64,10 @@
     "django.contrib.staticfiles",
     "django_extensions",
 
-    'home',
-    'theme_pixel'
+     'home',
+    'theme_pixel',
+    'rest_framework',
```
Comment (jbm1412) [inline core/settings.py:69]: Please add djangorestframework and its version in requirements.txt file

File: core/settings.py [hunk 2]
```diff
@@ -64,8 +64,10 @@
     "django.contrib.staticfiles",
     "django_extensions",
 
-    'home',
-    'theme_pixel'
+     'home',
+    'theme_pixel',
+    'rest_framework',  
+    'drf_yasg',
```
Comment (jbm1412) [inline core/settings.py:70]: Please add drf-yasg and its version in the requirements.txt file

**Review by jbm1412 — CHANGES_REQUESTED** [review]
Make these changes!

**Comment by Alankrita08** [comment]
Added `djangorestframework` and `drf-yasg` to requirements.txt with their respective versions. Resolved merge conflicts in `home/urls.py`.

File: core/settings.py (hunk 1 above)
Comment (Alankrita08) [reply core/settings.py:69]: image

File: core/settings.py (hunk 2 above)
Comment (Alankrita08) [reply core/settings.py:70]: image

**Review by s-bonnor — APPROVED** [review]
Looks good

**Review by jbm1412 — APPROVED** [review]

**Pull request merged**